### PR TITLE
Add IT for kv property source

### PR DESCRIPTION
**Pull request by saragluna**
# Description

Since Spring's integration tests, we have already provisioned the KV resources, but only tested the KV clients instead of the KV property source. This PR will enable the tests for KV property source. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/tests.yml [hunk 1]
```diff
@@ -14,10 +14,10 @@ stages:
           SubscriptionConfiguration: $(sub-config-cn-test-resources)
           Location: 'chinanorth3'
       TestResourceDirectories:
+        - spring/spring-cloud-azure-integration-tests/test-resources/keyvault
         - spring/spring-cloud-azure-integration-tests/test-resources/appconfiguration
         - spring/spring-cloud-azure-integration-tests/test-resources/cosmos
         - spring/spring-cloud-azure-integration-tests/test-resources/eventhubs
-        - spring/spring-cloud-azure-integration-tests/test-resources/keyvault
```
Comment (chenrujun) [inline sdk/spring/tests.yml:20]: Original order (alphabet order) is better.

**Review by chenrujun — APPROVED** [review]

**Review by Netyyyy — APPROVED** [review]
LGTM.

**Pull request merged**